### PR TITLE
do not layout `superview.delegate` from presentation layer

### PR DESCRIPTION
**Pull request by michaelknoch**
**Type of change:** performance improvement

## Motivation (current vs expected behavior)
This is a follow up based on https://github.com/flowkey/UIKit-cross-platform/pull/328.

I noticed that the recursive relayouting only happens when wrapping the layout code in an animation block. During this animation in the learnstep we mutate bounds of the presentation on every frame which then triggers layouting for the parent view for each frame as well. I think this matches the ios behaviour.

## Please check if the PR fulfills these requirements
- [ ] Self-review: I am confident this is the simplest and clearest way to achieve the expected behaviour
- [ ] There are no dependencies on other PRs or I have linked dependencies through Zenhub
- [ ] The commit messages are clean and understandable
- [ ] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

**Review by ephemer — APPROVED** [review]
It would be great if you could add a test that also runs on iOS to verify this behaviour but it seems plausible

File: UIKitTests/CALayerTests.swift [hunk 1]
```diff
@@ -69,4 +69,88 @@ class CALayerTests: XCTestCase {
...
+
+        let superView = TestView()
+        let view = TestView()
+        superView.addSubview(view)
+        view.layer.bounds.size = CGSize(width: 10, height: 10)
+
+        superView.layoutIfNeeded()
+        view.layoutIfNeeded()
+
+        view.layer.bounds.size = CGSize(width: 15, height: 15)
```
Comment (michaelknoch) [inline UIKitTests/CALayerTests.swift:94]: interestingly it does also relayout the superlayer when mutating `view.layer.bounds.origin`, but I think we should not mimic this behaviour because this also means that we relayout superlayers of UIScrollViews on every scrolled pixel which is not the case on iOS! (see the test below).

There seems to be a different connection between `contentOffset` and `bounds.origin` between ios and our UIKit 

@ephemer

Comment (ephemer) [reply UIKitTests/CALayerTests.swift]: I think we'd need to discuss this point on a call – not entirely sure what you mean about the scroll view right now. Is it possible to prevent the infinite layouts on the scrollview by changing that layer's delegate somehow?

I thought we only do a re-layout when the `bounds.size` changes though, so I don't understand how `contentOffset` or `bounds.origin` are related?

File: Sources/CALayer.swift [hunk 2]
```diff
@@ -133,7 +133,7 @@ open class CALayer {
             guard newBounds != bounds else { return }
             onWillSet(keyPath: .bounds)
 
-            if bounds.size != newBounds.size {
+            if !isPresentationForAnotherLayer && bounds.size != newBounds.size {
```
Comment (ephemer) [inline Sources/CALayer.swift:136]: how is this related to `contentOffset` @michaelknoch? That's the bit I don't understand right now

File: UIKitTests/CALayerTests.swift (hunk 1 above)
Comment (michaelknoch) [reply UIKitTests/CALayerTests.swift]: Setting contentOffset directly changes the bounds.origin of the scrollviews layer. On iOS and in our uikit - but I agree, could be more efficient to discuss this in a call. I'll send out an invite for next week.

- Anyways, the pr is ready for being merged. But I would like to share my findings about scrolllview bounds with you

Comment (ephemer) [reply UIKitTests/CALayerTests.swift]: @michaelknoch the behaviour we're talking about is only related to `bounds.size`, no? How is that related to bounds.origin / contentOffset?

File: Sources/CALayer.swift (hunk 2 above)
Comment (michaelknoch) [reply Sources/CALayer.swift:136]: on iOS this Check does not exist. They seem to always relayout the superlayer when bounds changes. 

But there is this exception where they do not relayout when setting the content offset of a scrollview.

This surprised me because from our understanding setting the contentOffset directly mutates the bounds.origin.


What does this mean? Initially
I thought we should remove the check and always relayout when bounds change (not only when bounds.size changes) but then we see this massive amount of relayouts when we scroll in our scrollview. Does this make sense @ephemer ? It's pretty hard to explain, it will be way easier to demonstrate.

Comment (ephemer) [reply Sources/CALayer.swift:136]: ahhhh ok sorry I didn't get the point about iOS also laying out the superview when `.origin` changes as well
